lxd/storage/backend_lxd.go

```diff
@@ -596,11 +608,23 @@ func (b *lxdBackend) SetCustomVolumeQuota(vol api.StorageVolume, quota uint64) e
 
 // MountCustomVolume mounts a custom volume.
 func (b *lxdBackend) MountCustomVolume(volName string, op *operations.Operation) (bool, error) {
+	unlock := lock(getVolumeMountLockID(volName))
```

Comment:
@stgraber @monstermunchkin its possible for custom volumes to be mounted passively via other operations, so if we need to lock them, this might be worth pushing down into the driver or the volume.MountTask function.

Reply:
Now for volumes, indeed there may be an issue if we don't do locking, so I think we can remove the locking from the backend (at least for now) and instead push it into the driver package and into MountTask as @tomponline suggested.

I think if we ever need locking in the backend, the preferred way for it will be to implement locking at the operations level (which is on my personal todo list) and just have the operations logic serialize identical operations.